### ISSUE TITLE
Times in list, why?

### ISSUE THREAD

**Issue by trojanobelix** [bramwaas/wordpress-plugin-wsa-simple-google-calendar-widget#3]
Hi,
Very nice Plugin!
I have a problem, that I always got times (23:00) in the list and I do not why,  DTSTART and DTEND do not have times
iCalendar Validator shows me no error, so it is a valid iCal file
[https://www.brochterbeck.de/wp-content/uploads/2020/11/IG_Brochterbeck.ics](url)
![image](https://user-images.githubusercontent.com/15106425/99849080-2a903e00-2b7b-11eb-9f6c-9b42e1e3a545.png)

BEGIN:VEVENT
UID:Ica241CF2AF5EEA4724974E1311C0707B79
LOCATION:St. Peter und Paul
SUMMARY:Neujahrsgottesdienst - Kath. Kirchengemeinde
DTSTART;VALUE=DATE:20200101
DTEND;VALUE=DATE:20200101
DTSTAMP;VALUE=DATE:20200101T000000Z
END:VEVENT
![image](https://user-images.githubusercontent.com/15106425/99849772-72639500-2b7c-11eb-98db-03bf94a28144.png)

btw: A reload button in the widget properties would be nicht to refresh the list data after upload a new iCal file(cache reload).

**Comment by bramwaas** [comment]
Hi Trojanobelix,

Thank you for your comment.

I see the issue.
The answer is that I always use a date time value and when time is not given I suppose 00:00:00 UTZ, that is in our timezone now 23:00:00 the previous day. But I understand that that is not always the behavior you would expect.

I hope I can find a solution and come back with a better version. Then I can also mark it comaptible to the current version of WP.

When that is finished I will think about the reaload button. I think it would indeed be an improvement, but in test I always used to save a small change as a work around. For instance change the cache time from 60 to 59 minutes  and save the widget. Then it also will refresh the cache.

Thanks and regards
Bram

**Comment by bramwaas** [comment]
@trojanobelix 
I think I solved the issue.
V1.2.0 available here on github, shortly I will also update it in WP plugins.
When an event has DATESTART as Date not DateTime, Time is not displayed.
An other error was that the the time in our timezone GMT +1 not was handled correct, bcause I used an older WP function date_i18n() was also solved by using the newer function wp_date().

Thanks and regards Bram

**Comment by trojanobelix** [comment]
Hi Bram,

thanks for the answer.

Other plugins have the option to disable time values for all events, even if the field is filled in the ics file and use an empty string (particular drop displaying time) if the current time field is empty (and time values enabled).

I was also wondering that the time is displayed twice, start time on the event and start to end time on the location.
I think it should be dropped at event line.

Btw. There are some empty events shown
[cid:image001.png@01D6C193.D107C930]

The ICS file does not contain empty fields as far as I can see. Maybe it is worth a look.  I could not find the reason, but my php efforts are very limited.

Again, thanks for your work.




Betreff: [NEWSLETTER] Re: [bramwaas/wordpress-plugin-wsa-simple-google-calendar-widget] Times in list, why? (#3)


Hi Trojanobelix,

Thank you for your comment.

I see the issue.
The answer is that I always use a date time value and when time is not given I suppose 00:00:00 UTZ, that is in our timezone now 23:00:00 the previous day. But I understand that that is not always the behavior you would expect.

I hope I can find a solution and come back with a better version. Then I can also mark it comaptible to the current version of WP.

When that is finished I will think about the reaload button. I think it would indeed be an improvement, but in test I always used to save a small change as a work around. For instance change the cache time from 60 to 59 minutes and save the widget. Then it also will refresh the cache.

Thanks and regards
Bram

**Comment by trojanobelix** [comment]
Thanks, that sounds good. I will get it a try.

Stay healthy.

Best wishes from Germany.

Der Obelix.



Betreff: [NEWSLETTER] Re: [bramwaas/wordpress-plugin-wsa-simple-google-calendar-widget] Times in list, why? (#3)


@trojanobelix<https://github.com/trojanobelix>
I think I solved the issue.
V1.2.0 available here on github, shortly I will also update it in WP plugins.
When an event has DATESTART as Date not DateTime, Time is not displayed.
An other error was that the the time in our timezone GMT +1 not was handled correct, bcause I used an older WP function date_i18n() was also solved by using the newer function wp_date().

Thanks and regards Bram

**Comment by bramwaas** [comment]
Hi TrojanObelix,

in V1.2.2 I have also made a checkbox to empty the cache. It will now only be emtied when that checkbox is checked (or when cache-time expired of course).

I will think about your suggestion to skip the times even when the iCal is in datetime. I was also thinking of making the times formatable just ike the date in the first line. An empty format would then completely omit the time.
The reason I have the start time in the line with the summary, is that the location is in my website only displayed when you click on the summary, using the javascript collapse function of the bootstrap framework. See attached picture (the green lines are links)
![trojanobelix](https://user-images.githubusercontent.com/11373174/100020957-fcac3300-2de0-11eb-8fbc-036842d9b790.png)


In my website I don't see the empty events (any more I don't know if the were presnt in the older version). I think the events in your  image are  'Bücherei - Kath. Kirchengemeinde' on 22 and 25 November. Special on these events is that they have no Location so they are earlier in the list for one event. 
I found that the UID's at the beginning of the event were not correct handled because I expected cr, lf (2 characters) end of lines and your file has only linefeeds (one character). It is possible that the solution for that was also a solution for the misinterpretation of Summaries but I can't explain why, anyway I hope it is solved.

Regards Bram.

**Comment by trojanobelix** [comment]
Thanks a lot, Bram, that sounds good. I will take a look on the link think with the collapse function. Seems nice.


Betreff: [NEWSLETTER] Re: [bramwaas/wordpress-plugin-wsa-simple-google-calendar-widget] Times in list, why? (#3)


Hi TrojanObelix,

in V1.2.2 I have also made a checkbox to empty the cache. It will now only be emtied when that checkbox is checked (or when cache-time expired of course).

I will think about your suggestion to skip the times even when the iCal is in datetime. I was also thinking of making the times formatable just ike the date in the first line. An empty format would then completely omit the time.
The reason I have the start time in the line with the summary, is that the location is in my website only displayed when you click on the summary, using the javascript collapse function of the bootstrap framework. See attached picture (the green lines are links)
[trojanobelix]<https://user-images.githubusercontent.com/11373174/100020957-fcac3300-2de0-11eb-8fbc-036842d9b790.png>

In my website I don't see the empty events (any more I don't know if the were presnt in the older version). I think the events in your image are 'Bücherei - Kath. Kirchengemeinde' on 22 and 25 November. Special on these events is that they have no Location so they are earlier in the list for one event.
I found that the UID's at the beginning of the event were not correct handled because I expected cr, lf (2 characters) end of lines and your file has only linefeeds (one character). It is possible that the solution for that was also a solution for the misinterpretation of Summaries but I can't explain why, anyway I hope it is solved.

Regards Bram.
